src/tools/fuzzing/fuzzing.cpp

```diff
@@ -270,6 +272,77 @@ void TranslateToFuzzReader::setupMemory() {
   }
 }
 
+void TranslateToFuzzReader::setupHeapTypes() {
+  // Start with any existing heap types in the module, which may exist in any
+  // initial content we began with.
+  auto possibleHeapTypes = ModuleUtils::collectHeapTypes(wasm);
+
+  // TODO: use heap type fuzzer to add new types in addition to the previous
+
+  // Filter away uninhabitable heap types, that is, heap types that we cannot
+  // construct, like a type with a non-nullable reference to itself. For
+  // simplicity, don't look for such type loops of aribtrary size but just limit
+  // the search to a reasonable amount (to avoid possible slowness). That is, we
+  // start at a given type, expand out its children, and continue to expand
+  // recursively. If there is a cycle then this would continue forever, so it
+  // must hit any fixed limit.
+  const size_t MAX_SEARCH = 100;
+  for (auto t : possibleHeapTypes) {
+    // These types are handled directly in the random code generators, and we
+    // assume they are not emitted by the code before us in this function.
+    assert(!t.isBasic() && !t.isBottom());
```

Comment:
This assertion you suggested actually gets hit. Specifically, I run `HeapTypeGenerator::create` and then `HeapTypeGenerator::makeInhabitable` and I get the heap type `none` in the vector.

Reply:
Oh yeah, the assertion is only meant to stick around until the next PR that uses `HeapTypeGenerator`.

Reply:
Oh, ok. I'll leave it here then.